indico/util/reporter.py

```diff
@@ -0,0 +1,72 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2016 European Organization for Nuclear Research (CERN).
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from copy import deepcopy
+
+from flask import session, request
+
+
+class ReporterBase(object):
+    """Serves as a base class for classes responsible to perform actions on
+    reports
+    """
+
+    #: A unique ID for the report
+    REPORT_ID = ''
+    #: Dict containing the default visible columns and filtering options
+    DEFAULT_REPORT_CONFIG = {}
+    #: OrderedDict containing the filterable columns of the report
+    FILTERABLE_ITEMS = None
+    cache = None
+
+    @classmethod
+    def get_report_config_session_key(cls, entry_parent_id):
+        """Composes the unique configuration ID that will be used as a key to
```

Comment:
FWIW, it's one of the guidelines we don't follow that much (command tense). Actually, "Returns ..." was my preference for quite some time. I wouldn't mind using the proper tense in new code though.

I fully agree with the second comment; having a short summary spares you the time of reading the detailed docstring. The disadvantage is that I've seen quite a few docstrings where it's almost the same as the function named (`s/_/ /` aside)